snap/validate.go

```diff
@@ -47,6 +57,15 @@ func Validate(info *Info) error {
 		return err
 	}
 
+	epoch := info.Epoch
+	if epoch == "" {
+		return fmt.Errorf("snap epoch cannot be empty")
```

Comment:
That does make it sound like the store is producing infos that are considered invalid by this function. This sounds like a bad precedent. Either we take this function seriously and do not accept handling invalid infos, or we change our idea of what it means for the info to be valid (effectively accepting empty epochs, in this particular case).

My vote would be to fix the store to ensure it does not produce invalid infos, but either way would be fine with me.